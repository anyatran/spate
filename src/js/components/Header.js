import React from 'react'
import PropTypes from 'prop-types'

const Header = ({ name }) => (
  <div className="header">
    <a href="/" className="header__logo">spate</a>
    <div className="header__profile">
      <a href="#" className="header__navigation">music</a>
      <a href="#" className="header__navigation">Hi {name}</a>
    </div>
  </div>
)

Header.propTypes = {
  name: PropTypes.string
}

export default Header;
