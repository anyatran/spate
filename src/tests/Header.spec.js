import React from 'react'
import { mount } from 'enzyme'
import Header from '../js/components/Header'

describe('<Header />', () => {
  it('should render Header', () => {
    const wrapper = mount(<Header name="anya"/>)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.header__navigation').length).toEqual(2)
  })
})
