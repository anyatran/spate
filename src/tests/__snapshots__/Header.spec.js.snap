// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header 1`] = `
<Header
  name="anya"
>
  <div
    className="header"
  >
    <a
      className="header__logo"
      href="/"
    >
      spate
    </a>
    <div
      className="header__profile"
    >
      <a
        className="header__navigation"
        href="#"
      >
        music
      </a>
      <a
        className="header__navigation"
        href="#"
      >
        Hi 
        anya
      </a>
    </div>
  </div>
</Header>
`;
